src/main/java/org/folio/innreach/specification/InnReachTransactionSpecification.java

```diff
@@ -0,0 +1,99 @@
+package org.folio.innreach.specification;
+
+import org.folio.innreach.domain.entity.InnReachTransactionSortingParameters;
+import org.folio.innreach.domain.entity.InnReachTransaction;
+import org.folio.innreach.domain.entity.TransactionItemHold;
+import org.folio.innreach.domain.entity.TransactionLocalHold;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+import static org.folio.innreach.domain.entity.InnReachTransaction.TransactionType.ITEM;
+import static org.folio.innreach.domain.entity.InnReachTransaction.TransactionType.LOCAL;
+
+@Component
+public class InnReachTransactionSpecification {
+
+  public Specification<InnReachTransaction> filterByParameters(InnReachTransactionSortingParameters parameters) {
+    return isOfType(parameters.getTypes())
+      .and(isOfState(parameters.getStates()))
+      .and(centralCodeIn(parameters.getCentralServerCodes()))
+      .and(patronAgencyIn(parameters.getPatronAgencyCodes()))
+      .and(itemAgencyIn(parameters.getItemAgencyCodes()))
+      .and(patronTypeIn(parameters.getPatronTypes()))
+      .and(centralItemTypeIn(parameters.getCentralItemTypes()));
+  }
+
+  static Specification<InnReachTransaction> isOfType(List<InnReachTransaction.TransactionType> types) {
+    return (transaction, cq, cb) -> {
+      if (Long.class != cq.getResultType()) {
```

Comment:
Could you please explain why this if statement need?
My understanding this method should perform comparison of transaction types, like:
transaction.type IN (<types>)
and this is done by the following statements

Reply:
We need this statement to correctly initialize count query for fetch joining `hold` and `pickupLocation` entities. Wasn't sure, where this check needs to be, so placed it here. Now moved it to a separate specification `fetchHoldAndPickupLocation()`.